projects/packages/forms/src/contact-form/class-contact-form.php

```diff
@@ -1741,4 +1741,95 @@ public static function get_block_alignment_class( $attributes = array() ) {
 		}
 		return $align_to_class_map[ $attributes['align'] ];
 	}
+
+	/**
+	 * Get the form's accessible name
+	 *
+	 * @param string $content The form's content.
+	 * @return string The form's accessible name.
+	 */
+	public function get_form_label( $content ) {
+		global $post;
+
+		$headings = array_map(
+			function ( $i ) {
+				return "h$i";
+			},
+			range( 1, 6 )
+		);
+
+		// 1. Look for a heading inside the form
+
+		$dom = new \DOMDocument();
+		$dom->loadHTML( $content );
+
+		foreach ( $headings as $heading ) {
+			$heading = $dom->getElementsByTagName( $heading )->item( 0 );
+
+			if ( isset( $heading ) ) {
+				$id = $heading->getAttribute( 'id' );
+
+				if ( isset( $id ) && ! empty( $id ) ) {
+					return "aria-labelledby='$id'";
+				}
+
+				return "aria-label='$heading->textContent'";
+			}
+		}
+
+		// 2. Look for a heading as a previous sibling
+
+		if ( has_blocks( $post->post_content ) ) {
+			$blocks = parse_blocks( $post->post_content );
+			// Filter out empty blocks
+			$blocks = array_values(
+				array_filter(
+					$blocks,
+					function ( $block ) {
+						return ! empty( $block['blockName'] );
+					}
+				)
+			);
+
+			foreach ( $blocks as $index => $block ) {
+				if ( $block['blockName'] !== 'jetpack/contact-form' ) {
+					continue;
+				}
+
+				$prev_block = $blocks[ $index - 1 ];
```

Comment:
Would this one give an error if `jetpack/contact-form` is the 1st in the `$blocks` array?